h2gis-utilities/src/main/java/org/h2gis/utilities/JDBCUtilities.java

```diff
@@ -868,6 +868,293 @@ public static String createTableDDL(ResultSet resultSet, String outputTableName)
+        if (connection == null || columnName == null || table == null) {
+            throw new SQLException("Unable to find an index");
+        }
+        boolean isH2 = isH2DataBase(connection);
+        if (isH2) {
+            PreparedStatement ps = connection.prepareStatement("SELECT INDEX_TYPE_NAME FROM INFORMATION_SCHEMA.INDEXES " +
+                    "WHERE INFORMATION_SCHEMA.INDEXES.TABLE_NAME=? " +
+                    "AND INFORMATION_SCHEMA.INDEXES.TABLE_SCHEMA=? " +
+                    "AND INFORMATION_SCHEMA.INDEXES.COLUMN_NAME=?;");
+            ps.setObject(1, table.getTable());
+            ps.setObject(2, table.getSchema("PUBLIC"));
+            ps.setObject(3, TableLocation.capsIdentifier(columnName, isH2));
+            ResultSet rs = ps.executeQuery();
+            while(rs.next()) {
+                if(rs.getString("INDEX_TYPE_NAME").contains("SPATIAL")){
+                    return true;
+                }
+            }
+            return false;
+        } else {
+            String query =  "SELECT  cls.relname, am.amname " +
+                    "FROM  pg_class cls " +
+                    "JOIN pg_am am ON am.oid=cls.relam where cls.oid " +
+                    " in(select attrelid as pg_class_oid from pg_catalog.pg_attribute " +
+                    " where attname = ? and attrelid in " +
+                    "(select b.oid from pg_catalog.pg_indexes a, pg_catalog.pg_class b  where a.schemaname =? and a.tablename =? " +
+                    "and a.indexname = b.relname)) and am.amname = 'gist' ;";
+            PreparedStatement ps = connection.prepareStatement(query);
+            ps.setObject(1, columnName);
+            ps.setObject(2, table.getSchema("public"));
+            ps.setObject(3, table.getTable());
+            ResultSet rs = ps.executeQuery();
+            return rs.next();
+        }
+    }
+
+    /**
+     * Create an index on the given column of the given table on the given connection.
+     * @param connection Connection to access to the desired table.
+     * @param table      Table containing the column to index.
+     * @param columnName Name of the column to index.
+     * @return True if the column have been indexed, false otherwise.
+     * @throws SQLException Exception thrown on SQL execution error.
+     */
+    public static boolean createIndex(Connection connection, TableLocation table, String columnName) throws SQLException {
+        if(connection == null || table == null || columnName == null){
+            throw new SQLException("Unable to create an index");
+        }
+        boolean isH2 = isH2DataBase(connection);
+        connection.createStatement().execute("CREATE INDEX IF NOT EXISTS " + table.toString(isH2) + "_" + columnName +
+                " ON " + table.toString(isH2) + " USING BTREE (" + TableLocation.capsIdentifier(columnName, isH2) + ")");
```

Comment:
Please, don't use `USING BTREE` in H2. It isn't documented and it isn't supported in any way. It may stop to work at any moment.

Reply:
Ok, I will use `CREATE SPATIAL INDEX` syntax, thank you.

Reply:
No, not here. `USING BTREE` (unlike `USING RTREE`) is no-op, it doesn't create a spatial index, it creates a regular index. The `SPATIAL` keyword should not be added here to preserve the old functionality.

Reply:
You can removed the using key word and do create index if not exists on tableName(columnName)